app/components/Settings/Preview/code-editor.jsx

```diff
@@ -0,0 +1,232 @@
+            CM.showHint(editor, CM.hint.sql, hintOptions);
+        };
+
+        this.editorDidMount = (editor) => {
+            this.editor = editor;
+
+            const wrapperElement = this.editor.getWrapperElement();
+            const height = wrapperElement.clientHeight;
+            const width = wrapperElement.clientWidth;
+
+            const minConstraints = [width, 74];
+            const maxConstraints = [width, Infinity];
+
+            this.setState({height, width, minConstraints, maxConstraints});
+        };
+
+        this.onResize = (event, data) => {
+            const height = data.size.height;
+
+            const wrapperElement = this.editor ? this.editor.getWrapperElement() : {};
+            const width = wrapperElement.clientWidth;
+
+            if (this.editor) this.editor.setSize(width, height);
+
+            this.setState({height, width});
+        };
+    }
+
+    static computeAutocompleteTables(props) {
+        const schemaRequest = props.schemaRequest || {};
+
+        if (schemaRequest.status !== 200 || !schemaRequest.content || !schemaRequest.content.rows) {
+            return {};
+        }
+
+        const codeSchema = {};
+        const bareTableNames = {};
+
+        schemaRequest.content.rows.forEach(function(row) {
+            const [tableName, columnName, dataType] = row;
+
+            let bareTableName = bareTableNames[tableName];
+            if (!bareTableName) {
+                bareTableName = tableName.split('.').slice(-1)[0];
+
+                const first = bareTableName.charAt(0);
+                if (first === '"' || first === '\'') {
+                    bareTableName = bareTableName.slice(1);
+                }
+
+                const last = bareTableName.charAt(bareTableName.length - 1);
+                if (last === '"' || last === '\'') {
+                    bareTableName = bareTableName.slice(0, -1);
+                }
+
+                if (!bareTableName) return;
+
+                bareTableNames[tableName] = bareTableName;
+            }
+
+            if (!codeSchema[bareTableName]) {
+                codeSchema[bareTableName] = {};
+            }
+
+            codeSchema[bareTableName][columnName] = dataType;
+        });
+
+        return codeSchema;
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.schemaRequest !== nextProps.schemaRequest) {
+            this.setState({
+                tables: CodeEditor.computeAutocompleteTables(this.props)
+            });
+        }
+    }
+
+    render() {
+        const {
+            editorDidMount,
+            onBeforeChange,
+            onChange,
+            onResize
+        } = this;
+
+        const {
+            value,
+            dialect,
+            runQuery,
+            isLoading
+        } = this.props;
+
+        const {
+            height,
+            width,
+            minConstraints,
+            maxConstraints
+        } = this.state;
+
+        const mode = {
```

Comment:
If we add other dialects should the be added here?  Should we update the NEW_CONNECTION doc?